### ISSUE TITLE
DateFieldInput: Extra announcement error message.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2465]
https://codesandbox.io/p/devbox/react-magma-example-forked-8mz563?file=%2Fexample.tsx%3A98%2C1&workspaceId=ws_UUfRFinNFpRPK24ozBo8Lb
Here is the example I put together. Try this:

Without entering anything try clicking Submit (you will hear Date is required - correct)
Enter some whacky date example: 04/04/0004. Click Submit (hear your error for proper year range - correct)
Now, use your datePicker to select correct date. For example 04/30/2026. Observe that reader doesn't show the error anymore for incorrect date but we still hear that error message.
Is there something we are missing here or is it something that is on Magma side that needs fixing. Thanks!

https://cengage.slack.com/archives/C0A9TSB75P0/p1777041799853289?thread_ts=1777040836.747919&cid=C0A9TSB75P0
https://cengage.slack.com/archives/C0A9TSB75P0/p1777040836747919

**Comment by artemtrusov-cengage** [comment]
### Tested on:

V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/datepicker--date-field-default
V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/datepicker--date-field-default

I tested it on Windows/macOS with NVDA/VoiceOver in Chrome/Safari. Everything works as expected. No bugs found.
